crev-lib/src/proofdb.rs

```diff
@@ -205,8 +206,10 @@ pub struct ProofDB {
     /// who -(trusts)-> whom
     trust_id_to_id: HashMap<Id, HashMap<Id, TimestampedTrustLevel>>,
 
-    url_by_id: HashMap<Id, TimestampedUrl>,
-    url_by_id_secondary: HashMap<Id, TimestampedUrl>,
+    verified_url_by_id: HashMap<Id, TimestampedUrl>,
```

Comment:
I would rename it to `url_by_id_self_reported`.

Reply:
So I think we have different cases here. My understanding is that crev has a distinction:

1. URLs that have been self reported
2. URLs that have been reported about someone else

But to me just these two options are not enough, because both allow URL spoofing in `lookup_url`, and that's a disaster for #211.

I know you don't put much value in securing the displayed URL, because you'd like crev to protect against GitHub/TLS hijacks, so these aren't supposed to be trusted anyway, but I'd like to up the security here from "anyone can show a fake URL just by signing whatever URL they want" to "attacker needs to actually take over control of the URL to spoof it". 

So in this PR I've changed it to:

1. URLs that have been self reported _and_ verified that the self-report is accurate.
2. URLs that have been reported about someone else, or self-reports that are unverified (and could be lies).

Given that (1) happens to be almost 100% of cases in normal usage, I wasn't worried about putting everything else in (2). But if you'd like to keep the original distinction too, then I could make it a 3-level set:

1. URLs that have been self-reported, and verified to be true (checked both ways: signature verifying Id->URL, and network check for URL->Id). These would be good for `cargo crev trust https://url`. 
2. URLs that have been self-reported, but may be lies (checked one way only).
3. URLs that have been reported about someone else (no checks at all).

Reply:
Yes. That clears it for me. Let's put it up in three levels like this, and please put a similar comments for each of them. Thank you for drilling it through my skull. I am indeed focused almost exclusively on starting from the ID case, and I agree that your change improves very much starting from the URL.